src/app/routes/utils/withRadioSchedule/index.js

```diff
@@ -0,0 +1,35 @@
+import { getRadioScheduleEndpoint } from '#lib/utilities/getRadioSchedulesUrls';
+import { fetchData } from '../fetchPageData';
+import { getQueryString } from '#lib/utilities/urlParser';
+import processRadioSchedule from '#containers/RadioSchedule/utilities/processRadioSchedule';
+
+const withRadioSchedules = async (pageDataPromise, service, path) => {
+  const { SIMORGH_APP_ENV, SIMORGH_BASE_URL } = process.env;
+
+  const radioScheduleUrl = getRadioScheduleEndpoint({
+    service,
+    env: SIMORGH_APP_ENV,
+    queryString: getQueryString(path),
+    baseUrl: SIMORGH_BASE_URL,
+  });
+
+  const radioSchedulePromise = fetchData(radioScheduleUrl);
```

Comment:
If we want to still render the page if this promise _rejects_ we can wrap this in error handling that resolves with no data, which will allow Promise.all to resolve if the page data is successfully fetched.